portfolio/src/main/java/com/google/sps/servlets/TeaVoteServlet.java

```diff
@@ -32,32 +33,69 @@
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //Update or add flavor entity.
         String flavor = request.getParameter("flavor");
+        Query flavorQuery = new Query("FlavorVote").setFilter(
+            new Query.FilterPredicate("flavor", Query.FilterOperator.EQUAL, flavor));
+        Entity flavorVoteEntity = datastore.prepare(flavorQuery).asSingleEntity();
+        long totalFlavorVotes = 0;
+
+        if (flavorVoteEntity != null) {
```

Comment:
this if statement is doing two unrelated things within it's branches, which is immediately a code smell to me. consider the following simplification:

if (flavorVoteEntity == null) {
  flavorVoteEntity = new Entity("FlavorVote");
  flavorVoteEntity.setProperty("flavor", flavor);
}

flavorVoteEntity.setProperty("votes", ((long) flavorVoteEntity.getProperty("votes")) + 1);

This is easier to read because there is less branching. It's simpler to understand because it's not combining two unrelated actions (creating the entity and getting the number of votes) into a single construct.

Reply:
I rearranged the null check so that the entity either gets updated or created within the if statement.